c++/src/OrcHdfsFile.cc

```diff
@@ -66,22 +64,22 @@ namespace orc {
         options = config->GetOptions();
       }
       hdfs::IoService * io_service = hdfs::IoService::New();
-      //Wrapping fs into a shared pointer to guarantee deletion
-      std::shared_ptr<hdfs::FileSystem> fs(hdfs::FileSystem::New(io_service, "", options));
-      if (!fs) {
+      //Wrapping file_system into a unique pointer to guarantee deletion
+      file_system = std::unique_ptr<hdfs::FileSystem>(hdfs::FileSystem::New(io_service, "", options));
+      if (!file_system) {
```

Comment:
I changed it here, however '!fs' notation is also used in many other parts of the libhdfs++ library, like examples, tests, tools, and c bindings, so hopefully this does not create too much trouble.